File: app/src/main/java/com/example/khj_pc/gaonnuri/Adapter/ChatRecyclerViewAdapter.java
package com.example.khj_pc.gaonnuri.Adapter;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.khj_pc.gaonnuri.Data.Chat;
import com.example.khj_pc.gaonnuri.R;

import java.util.ArrayList;

public class ChatRecyclerViewAdapter extends RecyclerView.Adapter<ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder> {

    private Context context;
    private ArrayList<Chat> chats;

    public ChatRecyclerViewAdapter(Context context, ArrayList<Chat> chats) {
        this.context = context;
        this.chats = chats;
    }

    @NonNull
    @Override
    public ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chat, parent, false);
        return new ChatRecyclerViewViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder holder, int position) {
        Chat chat = chats.get(position);
        holder.root.setBackgroundColor(Color.parseColor(position % 2 == 0 ? "#f5f9ff" : "#ffffff"));
        holder.username.setText(chat.getUserName());
        holder.time.setText(chat.getTime());
        holder.content.setText(chat.getContent());
    }

    @Override
    public int getItemCount() {
        return chats.size();
    }

    public static class ChatRecyclerViewViewHolder extends RecyclerView.ViewHolder {

        View root;
        TextView username, time, content;


        public ChatRecyclerViewViewHolder(@NonNull View itemView) {
            super(itemView);
            root = itemView;
            username = itemView.findViewById(R.id.item_chat_username);
            time = itemView.findViewById(R.id.item_chat_time);
            content = itemView.findViewById(R.id.item_chat_content);

        }

    }

}

File: app/src/main/java/com/example/khj_pc/gaonnuri/Adapter/ChatRecyclerViewAdapter.kt
package com.example.khj_pc.gaonnuri.Adapter

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.example.khj_pc.gaonnuri.Data.Chat
import com.example.khj_pc.gaonnuri.R
import com.example.khj_pc.gaonnuri.Util.SharedPreferenceUtil
import kotlinx.android.synthetic.main.item_chat.view.*

import java.util.ArrayList

class ChatRecyclerViewAdapter(private val context: Context, var chats: ArrayList<Chat>) : RecyclerView.Adapter<ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_chat, parent, false)
        return ChatRecyclerViewViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: ChatRecyclerViewAdapter.ChatRecyclerViewViewHolder, position: Int) {
        val chat = chats[position]
        val name : String = SharedPreferenceUtil.getPreference(context, "name")!!
        holder.root.setBackgroundColor(Color.parseColor(if (chat.userName == name) "#f5f9ff" else "#ffffff"))
        holder.root.item_chat_username.text = chat.userName
        holder.root.item_chat_time.text = chat.time
        holder.root.item_chat_content.text = chat.content
    }

    override fun getItemCount(): Int {
        return chats.size
    }

    fun addData(chat : Chat) {
        chats.add(chat)
        notifyDataSetChanged()
    }

    class ChatRecyclerViewViewHolder(internal var root: View) : RecyclerView.ViewHolder(root)

}
